### PR TITLE
feat(api): add dataloader to avoid refetching db

### PR DESCRIPTION
**Pull request by Tommytrg**

### CODE REVIEW

File: packages/api/src/loaders/index.ts [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+export class Loaders {
+  repositories: Repositories
+
+  constructor (repositories: Repositories) {
+    this.repositories = repositories
+  }
+  // returns a loader that fetches data using the given function
+  private genericLoader<T> (load: (filter) => T) {
+    return new DataLoader(async (filters: Array<string>) => {
+      console.log('here')
```
Comment (gabaldon) [inline packages/api/src/loaders/index.ts:13]: Delete console.log

File: packages/api/src/repository/ResultRequest.ts [hunk 2]
```diff
@@ -83,7 +83,8 @@ export class ResultRequestRepository {
         }
       }
     )
-    return this.normalizeId(lastResultRequest)
+
+    return lastResultRequest ? this.normalizeId(lastResultRequest) : null
```
Comment (gabaldon) [inline packages/api/src/repository/ResultRequest.ts:87]: Why is this necessary?

**Review by gabaldon — CHANGES_REQUESTED** [review]

File: packages/api/src/repository/ResultRequest.ts (hunk 2 above)
Comment (Tommytrg) [reply packages/api/src/repository/ResultRequest.ts:87]: Because if `lastRequestResult` isn't found, it can't be normalized

**Review by gabaldon — APPROVED** [review]
